File: PWRSyllabus.Core/Entities/StudyProgram.cs
using System.Collections.Generic;
using PWRSyllabus.Core.Enums;

namespace PWRSyllabus.Core.Entities
{
    public class StudyProgram : BaseEntity
    {
        public string Description { get; set; }
        public FormOfStudies FormOfStudies { get; set; }
        public Profile Profile { get; set; }
        public Level Level { get; set; }
        public string LanguageOfStudy { get; set; }
        public FieldOfStudy FieldOfStudy { get; set; }
        public int FieldOfStudyId { get; set; }

        public StudyPlan StudyPlan { get; set; }

        public List<StudyProgramSubjectCard> StudyProgramSubjectCards { get; set; }
    }
}

File: PWRSyllabus.API/ClientApp/src/models/StudyProgram.ts
import FieldOfStudy, { DefaultFieldOfStudy } from "./FieldOfStudy";
import SubjectCard from './SubjectCard';

export default interface StudyProgram {
    id: number,
    fieldOfStudy: FieldOfStudy;
    level: string;
    language: string;
    formOfStudies: string;
    subjectCards: SubjectCard[];
}

const DefaultStudyProgram: StudyProgram = {
    id: 0,
    fieldOfStudy: DefaultFieldOfStudy,
    level: '',
    language: '',
    formOfStudies: '',
    subjectCards: []
};

export {DefaultStudyProgram, StudyProgram};
